app/src/main/java/com/example/d3companion/presenters/list/ListPresenter.kt

```diff
@@ -1,21 +1,27 @@
 package com.example.d3companion.presenters.list
 
+import android.util.Log
 import com.example.d3companion.models.D3ViewElement
 import com.example.d3companion.models.D3ViewType
+import com.example.d3companion.services.D3IcyVeinsListener
+import com.example.d3companion.services.FileD3IcyVeinsProvider
 import com.example.d3companion.views.list.ListView
 import java.lang.ref.WeakReference
 import java.util.ArrayList
 
-class ListPresenter(view: ListView) : ListPresenterProvider {
+class ListPresenter(view: ListView, provider: FileD3IcyVeinsProvider) : ListPresenterProvider, D3IcyVeinsListener {
```

Comment:
Argument `provider` should be of type `D3IcyVeinsProvider` (the interface) and not `FileD3IcyVeinsProvider` (the implementation). This also implies that a property `listener` in the interface is needed.